src/Serialization/ToricGeometry.jl

```diff
@@ -43,3 +43,36 @@ function load_object(s::DeserializerState, ::Type{ToricDivisor}, tv::NormalToric
   pmdiv = Polymake._lookup_multi(pm_object(tv), "DIVISOR", index-1)
   return ToricDivisor(pmdiv, tv, coeffs)
 end
+
+################################################################################
+# Torus invariant divisor classes on toric varieties
+@register_serialization_type ToricDivisorClass uses_params
+
+function save_type_params(s::SerializerState, obj::ToricDivisorClass)
+  save_data_dict(s) do
+    save_object(s, encode_type(ToricDivisorClass), :name)
+    save_typed_object(s, obj.toric_variety, :params)
```

Comment:
It might be that toric varieties are stored with ids
```suggestion
      toric_ref = save_as_ref(s, obj.toric_variety)
      save_object(s, toric_ref, :params)
```

Reply:
Thank you @antonydellavecchia. But this comment should go into https://github.com/oscar-system/Oscar.jl/pull/3931.

The failures come from the changes in the 2nd commit, where I try to use the existing dict serializations.